### DISCUSSION TITLE
Are 4-Node Connectors Possible?

### DISCUSSION THREAD

**Discussion by ArturMakly** [xyflow/xyflow#1331] [Q&A] [answered]
hi, i noticed that in all your examples there are only 2 connectors on each node.
How can I add 2 more on the other sides?

example: 
![4-connectors](https://user-images.githubusercontent.com/40069318/125000696-56d3ce80-e027-11eb-802b-10bf8cc0914e.png)

Also... each connector needs to have the ability to have **_multiple_** edges attached to it.

![multi-node-connects](https://user-images.githubusercontent.com/40069318/125175314-a2aa8300-e1a1-11eb-8e2d-6e566dfd3df2.png)

**Comment by moklick** [comment, marked as answer]
Hey @ArturMakly 

that is possible with a custom node. You can find an example here: https://github.com/wbkd/react-flow/blob/main/example/src/Undirectional/CustomNode.tsx

**Reply by ArturMakly** [reply]
Thank you!!!!

**Reply by ArturMakly** [reply]
@moklick just realized that we will need to create a new `type` of node that is both a `Selector` && `Target`
easy to do??

**Reply by datoslabs** [reply]
@ArturMakly, could you elaborate on what a "node that is both a `Selector` && `Target`" means?

**Reply by ArturMakly** [reply]
sure! its’s basically the diagram you see above, whereby any of the 4 node-dots can receive and emit a connection edge/line to any other node ( multiple times ) 

here is another flow map that illustrates this: 
![](https://global-uploads.webflow.com/5d9d02c4b01a536c7d5d4a35/5dc307f8135ee9fd6debb383_flowchart-examples-4%402x.png)

**Reply by datoslabs** [reply]
Yes, they are controlled by `<Handle type="source".../>`, "source" means it can emit an outgoing connection, "target" means it can receive an incoming connection.

**Reply by ArturMakly** [reply]
@datoslabs so basically this will solve it right? 
https://github.com/wbkd/react-flow/discussions/1537#discussioncomment-1725776

**Reply by datoslabs** [reply]
Yes, just make sure to give each `handle` a unique id and specify it in the `edge` like the example.

**Reply by ArturMakly** [reply]
thank you!!

**Comment by cliffordfajardo** [comment]
I was also struggling with setting up custom nodes and making connections with it.

The example below is not the same example that @ArturMakly shared, but its similar & I'm sharing as it might help another person who stumbles upon this:

### Example
- I have a canvas with all custom nodes
- I want one node in particular to connect connect to multiple nodes

![CleanShot 2021-12-22 at 12 37 52](https://user-images.githubusercontent.com/6743796/147152370-792bf575-237b-4b17-942e-ef9b18e94be6.png)

CodeSandbox (`react-flow-renderer` version 9.7.1):
- https://codesandbox.io/s/nice-curran-s21yf?file=/src/examples/Example3/index.tsx

Thanks @datoslabs  lab for explaining it like this!
> Yes, they are controlled by <Handle type="source".../>, "source" means it can emit an outgoing connection, "target" means it can receive an incoming connection.
